packages/workflow-core/src/lib/plugins/external-plugin/bank-account-verification-plugin/bank-account-verification-plugin.ts

```diff
@@ -0,0 +1,221 @@
+});
+
+type TBankAccountVerificationPluginPayload = {
+  clientId: PluginPayloadProperty;
+  vendor: PluginPayloadProperty;
+  data: {
+    address: {
+      streetNumber: PluginPayloadProperty;
+      street: PluginPayloadProperty;
+      city: PluginPayloadProperty;
+      postcode: PluginPayloadProperty;
+    };
+    bankAccountDetails: {
+      sortCode: PluginPayloadProperty;
+      bankAccountNumber: PluginPayloadProperty;
+    } & (
+      | {
+          holder: {
+            firstName: PluginPayloadProperty;
+            middleName: PluginPayloadProperty<string | undefined>;
+            lastName: PluginPayloadProperty;
+          };
+        }
+      | ({ bankAccountName: PluginPayloadProperty } & (
+          | { companyRegistrationNumber: PluginPayloadProperty }
+          | { registeredCharityNumber: PluginPayloadProperty }
+        ))
+    );
+  };
+};
+
+export class BankAccountVerificationPlugin extends ApiPlugin {
+  public static pluginType = 'http';
+  public payload: TBankAccountVerificationPluginPayload;
+
+  private pluginName = 'Bank Account Verification Plugin';
+
+  constructor({
+    payload,
+    ...pluginParams
+  }: IApiPluginParams & { payload: BankAccountVerificationPlugin['payload'] }) {
+    super({
+      ...pluginParams,
+      response: {
+        ...pluginParams.response,
+        transformers: [
+          ...(pluginParams.response?.transformers ?? []),
+          getTransformer({
+            mapping: [
+              {
+                method: 'setTimeToRecordUTC',
+                source: 'invokedAt',
+                target: 'invokedAt',
+              },
+            ],
+            transformer: 'helper',
+          }),
+        ],
+      },
+      method: 'POST' as const,
+    });
+
+    this.payload = payload;
+
+    handleJmespathTransformers({
+      pluginName: this.pluginName,
+      requestTransformers: this.request?.transformers,
+      responseTransformers: this.response?.transformers,
+    });
+  }
+
+  async invoke(context: TContext) {
+    const env = validateEnv(this.pluginName);
+
+    let requestPayload;
+
+    if (this.request?.transformers) {
+      requestPayload = await this.transformData(this.request.transformers, context);
+
+      const { isValidRequest, errorMessage } = await this.validateContent(
+        this.request.schemaValidator,
+        requestPayload,
+        'Request',
+      );
+
+      if (!isValidRequest) {
+        return this.returnErrorResponse(errorMessage ?? 'Invalid request');
+      }
+    }
+
+    try {
+      const url = `${env.UNIFIED_API_URL}/bank-account-verification/commercial`;
+
+      const payload = getPayloadPropertiesValue({
+        properties: this.payload,
+        context,
+      });
+
+      const validatedPayload = BankAccountVerificationPluginPayloadSchema.safeParse(payload);
+
+      if (!validatedPayload.success) {
+        return this.returnErrorResponse(
+          `Invalid payload: ${JSON.stringify(validatedPayload.error.errors)}`,
```

Comment:
Add the plugin name - can be a static class variable - maybe we should also enforce this pattern (having a name)